llvm/lib/Transforms/Coroutines/CoroShape.h

```diff
@@ -50,15 +50,49 @@ enum class ABI {
 // Holds structural Coroutine Intrinsics for a particular function and other
 // values used during CoroSplit pass.
 struct LLVM_LIBRARY_VISIBILITY Shape {
-  CoroBeginInst *CoroBegin;
+  CoroBeginInst *CoroBegin = nullptr;
   SmallVector<AnyCoroEndInst *, 4> CoroEnds;
   SmallVector<CoroSizeInst *, 2> CoroSizes;
   SmallVector<CoroAlignInst *, 2> CoroAligns;
   SmallVector<AnyCoroSuspendInst *, 4> CoroSuspends;
-  SmallVector<CallInst *, 2> SwiftErrorOps;
   SmallVector<CoroAwaitSuspendInst *, 4> CoroAwaitSuspends;
   SmallVector<CallInst *, 2> SymmetricTransfers;
 
+  // Values invalidated by invalidateCoroutine() and cleanCoroutine()
+  SmallVector<CoroFrameInst *, 8> CoroFrames;
+  SmallVector<CoroSaveInst *, 2> UnusedCoroSaves;
```

Comment:
nit: The name looks slightly odd to me. It looks like they will be cleared in invalidateCoroutine(). This is conflicting with the comment.

And also I like to rename them to `ToBeInvalidatedCoroFrames` and so on. Then in `analyze`, remain the original `CoroFrames` and `UnusedCoroSaves`, let's only assign values to `ToBeInvalidatedCoroFrames` (and so on) when it is invalid actually.

Reply:
I see you approved the change but I don't see a lgtm? Do you want me to make these nit changes in another patch?

About the nit changes, let me see if I have understood what you want.

1. Keep CoroShape::CoroFrames and CoroShape::UnusedCoroSaves
2. Declare CoroShape::ToBeInvalidatedCoroFrames and CoroShape::ToBeInvalidatedUnusedCoroSaves
3. In CoroShape::invalidateCoroutine() and CoroShape::cleanCoroutine() instead of erasing the Instructions, move them to CoroShape::ToBeInvalidated{CoroFrames,UnusedCoroSaves}
4. Add code to CoroShape::clear() or CoroShape::~CoroShape() to erase the Instructions in CoroShape::ToBeInvalidated{CoroFrames,UnusedCoroSaves}

Is that right?

Should we erase instructions in CoroShape::clear() or CoroShape::~CoroShape()?

Also I noticed that CoroShape::invalidateCoroutine() also erases CoroSuspends and CoroSaves (those that are used). Should we also delay erasing these until CoroShape::clear() or CoroShape::~CoroShape()?

Reply:
My original intent to the changes was to ensure that we could create CoroShape objects from the Function without changing the Function's instructions in anyway. This makes CoroShape a little more useful because right now if a method needs the info we have to rely on it being passed into the method, and it cannot be used by other passes.

Reply:
> I see you approved the change but I don't see a lgtm? Do you want me to make these nit changes in another patch?

Approve with a nit means, you can land it as if now, but I'll be more happier if you address the nit comments.

> 1. Keep CoroShape::CoroFrames and CoroShape::UnusedCoroSaves

I mean, keep them as is, a local variables, IIUC.

> My original intent to the changes was to ensure that we could create CoroShape objects from the Function without changing the Function's instructions in anyway. 

The idea sounds good. But it will erase the instructions, right? I don't feel bad with this since this is the current behavior of the building process of CoroShape. But it will be good to make it a pure analysis process.

Reply:
Okay, I can see I made an unintentional functional change to CoroShape (changing the building process). I will make CoroFrames and UnusedCoroSaves local variables again. I have some ideas about how to make a pure analysis. I will pursue that after landing the ABI object patches.